lib/pdk/generate/module.rb

```diff
@@ -68,19 +68,21 @@ def self.invoke(opts = {})
           metadata.write!(File.join(temp_target_dir, 'metadata.json'))
         end
 
-        if template_url == PDK::Util.puppetlabs_template_url
-          # If the user specifies our template via the command line, remove the
-          # saved template-url answer.
+        # Only update the answers files after metadata has been written.
+        if template_uri == PDK::Util.default_template_uri
+          # If the user specifies our default template url via the command
+          # line, remove the saved template-url answer so that the template_uri
+          # resolution can find new default URLs in the future.
           PDK.answers.update!('template-url' => nil) if opts.key?(:'template-url')
         else
-          # Save the template-url answer if the module was generated using
-          # a template other than ours.
-          PDK.answers.update!('template-url' => template_url)
+          # Save the template-url + template-ref answers if the module was
```

Comment:
`template-ref` is not being saved to the answers

Reply:
That's by design, so I'll update the comment. 

To expound on that, the git reference is encoded into the `template-url` via the fragment. The only time `template-ref` is externalized as part of pdk is when it's added to the metadata.json to indicate the last template ref that was used by `pdk update`.